libraries/networking/src/udt/SendQueue.cpp

```diff
@@ -63,6 +64,46 @@ void SendQueue::queuePacket(std::unique_ptr<Packet> packet) {
     }
 }
 
+void SendQueue::queuePacketList(std::unique_ptr<PacketList> packetList) {
+    Q_ASSERT(packetList->_packets.size() > 0);
+
+    {
+        auto messageNumber = getNextMessageNumber();
+
+        if (packetList->_packets.size() == 1) {
+            auto packet = packetList->_packets.front().get();
+
+            packet->setPacketPosition(Packet::PacketPosition::ONLY);
+            packet->writeMessageNumber(messageNumber);
+        } else {
+            bool haveMarkedFirstPacket = false;
+            auto end = packetList->_packets.end();
+            for (auto it = packetList->_packets.begin(); it != end; ++it) {
+                auto packet = it->get();
+
+                if (!haveMarkedFirstPacket) {
+                    packet->setPacketPosition(Packet::PacketPosition::FIRST);
+                    haveMarkedFirstPacket = true;
+                } else if (packetList->_packets.empty()) {
+                    packet->setPacketPosition(Packet::PacketPosition::LAST);
```

Comment:
Good catch - we should check to see if we are at the end of the list via the iterator